tempesta_fw/http_msg.c

```diff
@@ -759,6 +759,74 @@ tfw_http_msg_del_hbh_hdrs(TfwHttpMsg *hm)
 	return 0;
 }
 
+/**
+ * Modify message body, add string @data to the end of the body (if @append) or
+ * to its beginning.
+ */
+static int
+tfw_http_msg_body_xfrm(TfwHttpMsg *hm, TfwStr *data, bool append)
+{
+	int r;
+	TfwStr it = {};
+	char *st = append
+		? (TFW_STR_CURR(&hm->body)->data + TFW_STR_CURR(&hm->body)->len)
+		: (TFW_STR_CHUNK(&hm->body, 0)->data);
+
+	r = ss_skb_get_room(hm->msg.skb_head, hm->body.skb,
```

Comment:
`hm->body.skb` is the first body skb, right? Shouldn't `ss_skb_peek_tail(&hm->msg.skb_head)` be there? What if there are trailing headers? Seems like a bug, fix if so. In any case a good comment is required here. The closed connection as an indicator of the response end is a rare case, so we don't care about performance now. But the code for response chunkining is useful for #498 streams - many responses are going to be chunked on our side. So it's crucial for us to wrie `<chunk_len>\r\n` at `skb->tail` to reuse the allocation overhead and avoid possible skb allocations. If we don't write the chunk length strictly at `skb->tail`, then leave TODO comment for #498.

Reply:
Thanks! there was a bug, I've fixed it. 

There can't be trailer headers. The message is being converted to chunked encoding, so it couldn't be in chunked encoding before. Since trailer may appear only after chunked body, it couldn't be present in the original message. I've added a comment to show this.